bundles/org.openhab.binding.sunsynk/src/main/java/org/openhab/binding/sunsynk/internal/api/AccountController.java

```diff
@@ -0,0 +1,195 @@
+                                // unsed block under null check
+    private void sendHttp(String payload) throws SunSynkAuthenticateException, SunSynkTokenException {
+        Gson gson = new Gson();
+        String response = "";
+        String httpsURL = makeLoginURL("oauth/token");
+        Properties headers = new Properties();
+        headers.setProperty("Accept", "application/json");
+        InputStream stream = new ByteArrayInputStream(payload.getBytes(StandardCharsets.UTF_8));
+        try {
+            response = HttpUtil.executeUrl(HttpMethod.POST.asString(), httpsURL, headers, stream, "application/json",
+                    TIMEOUT_IN_MS);
+
+            @Nullable
+            Client client = gson.fromJson(response, Client.class);
+            if (client == null) {
+                throw new SunSynkAuthenticateException(
+                        "Sun Synk Account could not be authenticated: Try re-enabling account");
+            }
+            this.sunAccount = client;
+        } catch (IOException | JsonSyntaxException e) {
+            throw new SunSynkAuthenticateException("Sun Synk Account could not be authenticated", e);
+        }
+        if (this.sunAccount.getCode() == 102) {
+            logger.debug("Sun Synk Account could not be authenticated: {}.", this.sunAccount.getMsg());
+            throw new SunSynkAuthenticateException(
+                    "Sun Synk Accountfailed to authenticate: Check your password or email.");
+        }
+        if (this.sunAccount.getStatus() == 404) {
+            logger.debug("Sun Synk Account could not be authenticated: 404 {} {}.", this.sunAccount.getError(),
+                    this.sunAccount.getPath());
+            throw new SunSynkAuthenticateException("Sun Synk Accountfailed to authenticate: 404 Not Found.");
+        }
+        getToken();
+    }
+
+    private void getToken() throws SunSynkAuthenticateException {
+        APIdata data = this.sunAccount.getData();
+        APIdata.staticAccessToken = data.getAccessToken();
+    }
+
+    /**
+     * Discovers a list of all inverter tied to a Sunsynk Connect Account
+     * 
+     * @return List of connected inveters
+     * @throws SunSynkInverterDiscoveryException
+     */
+    @SuppressWarnings("unused")
+    public ArrayList<Inverter> getDetails() throws SunSynkInverterDiscoveryException {
+        Details output = new Details();
+        ArrayList<Inverter> inverters = new ArrayList<>();
+        try {
+            Gson gson = new Gson();
+            Properties headers = new Properties();
+            String response = "";
+            String httpsURL = makeLoginURL(
+                    "api/v1/inverters?page=1&limit=10&total=0&status=-1&sn=&plantId=&type=-2&softVer=&hmiVer=&agentCompanyId=-1&gsn=");
+            headers.setProperty("Accept", "application/json");
+            headers.setProperty("Authorization", "Bearer " + APIdata.staticAccessToken);
```

Comment:
Prefer that constants are used instead of magic strings:
https://docs.oracle.com/javaee/7/api/javax/ws/rs/core/MediaType.html#APPLICATION_JSON
https://docs.oracle.com/javaee/6/api/javax/ws/rs/core/HttpHeaders.html#ACCEPT
https://docs.oracle.com/javaee/6/api/javax/ws/rs/core/HttpHeaders.html#AUTHORIZATION
Bearer can probably be found elsewhere.

Reply:
Done